homeassistant/components/notify/pushsafer.py

```diff
@@ -5,20 +5,49 @@
 https://home-assistant.io/components/notify.pushsafer/
 """
 import logging
-
+import base64
+import mimetypes
 import requests
+from requests.auth import HTTPBasicAuth
 import voluptuous as vol
 
 from homeassistant.components.notify import (
-    ATTR_TITLE, ATTR_TITLE_DEFAULT, PLATFORM_SCHEMA, BaseNotificationService)
+    ATTR_TITLE, ATTR_TITLE_DEFAULT, ATTR_TARGET, ATTR_DATA,
+    PLATFORM_SCHEMA, BaseNotificationService)
 import homeassistant.helpers.config_validation as cv
 
 _LOGGER = logging.getLogger(__name__)
 _RESOURCE = 'https://www.pushsafer.com/api'
+_ALLOWED_IMAGES = ['image/gif', 'image/jpeg', 'image/png']
 
 CONF_DEVICE_KEY = 'private_key'
+CONF_TIMEOUT = 15
+
+# Top level attributes in 'data'
+ATTR_SOUND = 'sound'
+ATTR_VIBRATION = 'vibration'
+ATTR_ICON = 'icon'
+ATTR_ICONCOLOR = 'iconcolor'
+ATTR_URL = 'url'
+ATTR_URLTITLE = 'urltitle'
+ATTR_TIME2LIVE = 'time2live'
+ATTR_PICTURE1 = 'picture1'
 
-DEFAULT_TIMEOUT = 10
+ATTR_SOUND_DEFAULT = ''
+ATTR_VIBRATION_DEFAULT = ''
+ATTR_ICON_DEFAULT = ''
+ATTR_ICONCOLOR_DEFAULT = ''
+ATTR_URL_DEFAULT = ''
+ATTR_URLTITLE_DEFAULT = ''
+ATTR_TIME2LIVE_DEFAULT = ''
+ATTR_PICTURE1_DEFAULT = ''
```

Comment:
Remove this defaults. In this case it's legitimate to have that on code

Reply:
@pvizeli @balloob 
I have completed your change requests